src/schemas.jl

```diff
@@ -344,6 +344,23 @@ abstract type AbstractRecord <: Tables.AbstractRow end
 @inline Tables.columnnames(r::AbstractRecord) = fieldnames(typeof(r))
 @inline Tables.schema(::AbstractVector{R}) where {R<:AbstractRecord} = Tables.Schema(fieldnames(R), fieldtypes(R))
 
+# we need a bit of extra work to integrate with ConstructionBase for pre-1.7 due
+# to the overload of `propertynames(::Tables.AbstractRow)`
+#
+# we _could_ overload `check_properties_are_fields` but that's no part of the
+# public API so this is safer
+@static if VERSION < v"1.7"
+    ConstructionBase.getproperties(r::AbstractRecord) = NamedTuple(r)
+    # largely copy-paste from ConstructionBase.setproperties_object:
```

Comment:
Can you provide permalinks to the code this is based off of? Makes this easier to maintain